content/en/blog/_posts/2020-09-30-writing-crl-scheduler/index.md

```diff
@@ -0,0 +1,97 @@
+
+Worse yet, our automation, at the time, would remove Nodes A-2, B-2, and C-2. Leaving CRDB-1 in an unscheduled state as persistent volumes are only available in the zone they are initially created in.
+
+To correct the latter issue, we now employ a "hunt and peck" approach to removing machines from a cluster. Rather than blindly removing Kubernetes nodes from the cluster, only nodes without a CockroachDB pod would be removed. The much more daunting task was to wrangle the Kubernetes scheduler.
+
+## A session of brainstorming left us with 3 options:
+
+### 1. Upgrade to kubernetes 1.18 and make use of Pod Topology Spread Constraints
+
+While this seems like it could have been the perfect solution, at the time of writing Kubernetes 1.18 was unavailable on  the two most common managed Kubernetes services in public cloud, EKS and GKE. Furthermore, [pod topology spread constraints](/docs/concepts/workloads/pods/pod-topology-spread-constraints/) were still a beta feature which meant that it wasn't guaranteed to be available in managed clusters even when v1.18 became available.
+The entire endeavour was concerningly reminiscent of checking [caniuse.com](https://caniuse.com/) when Internet Explorer 8 was still around.
+
+### 2. Deploy a statefulset _per zone_.
+
+Rather than having one StatefulSet distributed across all availability zones, a single StatefulSet with node affinities per zone would allow manual control over our zonal topology.
+Our team had considered this as an option in the past which made it particularly appealing.
+Ultimately, we decided to forego this option as it would have required a massive overhaul to our codebase and performing the migration on existing customer clusters would have been an equally large undertaking.
+
+### 3. Write a custom Kubernetes scheduler.
+
+(https://kubernetes.io/docs/tasks/extend-kubernetes/configure-multiple-schedulers/)
+
+At its core, this entire issue was a misunderstanding with the guarantees that kube-scheduler provided us with. Why not provide ourselves with the original guarantee that we were looking for?
+
+Thanks to an example from [Kelsey Hightower](https://github.com/kelseyhightower/scheduler) and a blog post from [Banzai Cloud](https://banzaicloud.com/blog/k8s-custom-scheduler/), we decided to dive in head first and write our own [custom Kubernetes scheduler](/docs/tasks/extend-kubernetes/configure-multiple-schedulers/). Once our proof-of-concept was deployed and running, we quickly discovered that the Kubernetes scheduler is also responsible for mapping persistent volumes to the Pods that it schedules, which wasn't clear to us from the [documentation](/docs/concepts/scheduling-eviction/) about scheduling, nor evident from running [`kubectl get events`](/docs/tasks/extend-kubernetes/configure-multiple-schedulers/#verifying-that-the-pods-were-scheduled-using-the-desired-schedulers).
```

Comment:
nit: I am not sure if this phrase adds much. Could you reword or update the section:
```
 which wasn't clear to us from the [documentation](/docs/concepts/scheduling-eviction/) about scheduling, nor evident from running [`kubectl get events`](/docs/tasks/extend-kubernetes/configure-multiple-schedulers/#verifying-that-the-pods-were-scheduled-using-the-desired-schedulers)
```

Reply:
Agreed! I've reworded it to show case that our confusion is what caused us to stumble upon the plugin system.